src/lib/containers/UserCard.tsx

```diff
@@ -62,95 +67,75 @@ export const UserCard: React.FunctionComponent<UserCardProps> = (
-    string | null | undefined
-  >(undefined)
+
+  // If we were given an alias, fetch the principal ID
+  const { data: fetchedPrincipalId, isLoading: isLoadingAliasLookup } =
+    useGetPrincipalIdForAlias(
+      {
+        alias: alias!,
+        type: AliasType.USER_NAME,
+      },
+      { enabled: !!alias },
+    )
+
+  const principalId = (
+    ownerId ??
+    initialProfile?.ownerId ??
+    fetchedPrincipalId
+  )?.toString()
 
   // If we weren't provided an initialProfile, fetch from Synapse
-  const { data: fetchedProfile } = useGetUserProfile(principalId!, {
-    enabled: !!principalId && !initialProfile,
-  })
+  const { data: fetchedProfile, isLoading: isLoadingProfile } =
+    useGetUserProfile(principalId!, {
+      enabled: !!principalId && !initialProfile,
+    })
   const userProfile = initialProfile ?? fetchedProfile
 
-  useEffect(() => {
-    let isCanceled = false
-    // If we weren't provided an inital presigned URL for the image, fetch from Synapse.
-    if (!initialPreSignedURL && (ownerId || userProfile?.ownerId)) {
-      SynapseClient.getProfilePicPreviewPresignedUrl(
-        (ownerId ?? userProfile?.ownerId)!,
-      ).then(result => {
-        if (!isCanceled) {
-          setFetchedPresignedUrl(result)
-        }
-      })
-    }
-    return () => {
-      isCanceled = true
-    }
-  }, [initialPreSignedURL, ownerId, userProfile?.ownerId])
+  const { data: avatarData, isLoading: isLoadingAvatar } = useGetProfileImage(
+    principalId!,
+    {
+      // Don't fetch the avatar if `initialPresignedURL` is provided.
+      // Also, wait until we have a value for `principalId`
+      enabled: !!(!initialPreSignedURL && principalId),
+    },
+  )
 
-  const presignedUrl = initialPreSignedURL ?? fetchedPresignedUrl
+  // If a URL was provided, fetch it and create a local data URL
+  const providedAvatarURL = usePreFetchResource(initialPreSignedURL)
+  // If we fetched the avatar from Synapse, we already have the blob. Create a local data URL
+  const fetchedAvatarUrl = useCreateUrlForData(avatarData)
 
-  const mayHaveProfileImage = presignedUrl !== null
-  const imageURL = usePreFetchResource(
-    mayHaveProfileImage ? presignedUrl : undefined,
-  )
-  const isLoadingAvatar = mayHaveProfileImage && !imageURL
+  const imageURL = providedAvatarURL ?? fetchedAvatarUrl
 
-  useEffect(() => {
-    let isCanceled = false
-    if (userProfile) {
-      setIsLoading(false)
-    } else if (alias) {
-      // Before we can get the profile, we must get the principal ID using the alias
-      getPrincipalAliasRequest(accessToken, alias, 'USER_NAME').then(
-        aliasData => {
-          if (!isCanceled) {
-            setPrincipalId(aliasData.principalId.toString())
-          }
-        },
-      )
-    }
-    return () => {
-      isCanceled = true
-    }
-  }, [userProfile, alias, accessToken])
+  const isLoading = isLoadingAliasLookup || isLoadingProfile
 
-  function Card(props: {
```

Comment:
This fn component was defined inside of the UserCard fn component, so we probably had excessive re-renders here.